src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal.tsx

```diff
@@ -0,0 +1,171 @@
+        const isCreateButtonDisabled = !values.name
+
+        return (
+          <ModalDialog
+            width={["100%", 713]}
+            onClose={onClose}
+            title={t("collectorSaves.createNewListModal.title")}
+            data-testid="CreateNewList"
+            footer={
+              <>
+                {/* Desktop view */}
+                <Media greaterThanOrEqual="sm">
+                  <Flex justifyContent="flex-end">
+                    <Button
+                      disabled={isCreateButtonDisabled}
+                      loading={isSubmitting}
+                      onClick={() => handleSubmit()}
+                    >
+                      {t("collectorSaves.createNewListModal.createList")}
+                    </Button>
+                  </Flex>
+                </Media>
+
+                {/* Mobile view */}
+                <Media lessThan="sm">
+                  <Button
+                    disabled={isCreateButtonDisabled}
+                    loading={isSubmitting}
+                    onClick={() => handleSubmit()}
+                    width="100%"
+                  >
+                    {t("collectorSaves.createNewListModal.createList")}
+                  </Button>
+
+                  <Spacer y={1} />
+
+                  <Button
+                    variant="secondaryBlack"
+                    width="100%"
+                    onClick={onClose}
+                  >
+                    {t("collectorSaves.createNewListModal.back")}
+                  </Button>
+                </Media>
+              </>
+            }
+          >
+            <LabeledInput
+              title={t("collectorSaves.createNewListModal.nameLabel")}
+              name="name"
+              placeholder={t(
+                "collectorSaves.createNewListModal.namePlaceholder"
+              )}
+              value={values.name}
+              onChange={e => {
+                handleChange("name")(e)
+                setRemainingCharacters(
+                  Math.max(maxNameLength - e.target.value.length, 0)
+                )
+                setFormFieldError("")
+              }}
+              onBlur={handleBlur("name")}
+              error={formFieldError}
+              maxLength={maxNameLength}
+              label={
+                <>
+                  {/* Desktop view */}
+                  <Media greaterThanOrEqual="sm">
+                    <EditIcon />
+                  </Media>
+                </>
+              }
+            />
+            <Spacer y={1} />
+            <Text variant="xs">
+              {t("collectorSaves.createNewListModal.remainingCharacters", {
+                count: remainingCharacters,
+              })}
+            </Text>
+          </ModalDialog>
+        )
+      }}
+    </Formik>
+  )
+}
+
+export const CreateNewListModalContainer: React.FC<CreateNewListModalProps> = props => {
+  const { visible } = props
```

Comment:
#suggestion - its always clearer to exit early if a condition is not met, vs rendering a component conditionally, and then having stuff below it:

```tsx
if (!visible) {
  return null
}

return <MyComponent />
```